### PR TITLE
Update upstream to release 0.7-1

### PR DESCRIPTION
**Pull request by richiware**
Tested in this [Fast-DDS PR](https://github.com/eProsima/Fast-DDS/pull/1996)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -68,15 +68,16 @@ if(NOT foonathan_memory_FOUND)
...
       git apply ${CMAKE_CURRENT_SOURCE_DIR}/CMakeLists.txt.patch &&
       sed -i -e "s/INSTALLER_PLATFORM/${INSTALLER_PLATFORM}/g" CMakeLists.txt)
-  elseif(BUILD_MEMORY_TESTS)
-    # This could be removed whenever we change the GIT_TAG from c619113 to anything above 09b884c
+  else()
+    # This could be removed whenever we change the GIT_TAG to anything above e9e13b0de82032051a49fd12ba2962e26949fb75
     set(PATCH_COMMAND_STR PATCH_COMMAND
-      curl https://github.com/foonathan/memory/commit/09b884c18abb314bfa678df27141a4dcad71c4ba.diff | git apply -)
+      curl https://github.com/MiguelCompany/memory/commit/6041b31341c323d07611051bb6e1970aabee5532.diff | git apply -)
+
```
Comment (MiguelCompany) [inline CMakeLists.txt:75]: The whole `else` could be removed

**Review by MiguelCompany — CHANGES_REQUESTED** [review]

File: CMakeLists.txt [hunk 2]
```diff
@@ -68,15 +68,16 @@ if(NOT foonathan_memory_FOUND)
     set(PATCH_COMMAND_STR PATCH_COMMAND
       git apply ${CMAKE_CURRENT_SOURCE_DIR}/CMakeLists.txt.patch &&
       sed -i -e "s/INSTALLER_PLATFORM/${INSTALLER_PLATFORM}/g" CMakeLists.txt)
-  elseif(BUILD_MEMORY_TESTS)
-    # This could be removed whenever we change the GIT_TAG from c619113 to anything above 09b884c
+  else()
+    # This could be removed whenever we change the GIT_TAG to anything above e9e13b0de82032051a49fd12ba2962e26949fb75
     set(PATCH_COMMAND_STR PATCH_COMMAND
-      curl https://github.com/foonathan/memory/commit/09b884c18abb314bfa678df27141a4dcad71c4ba.diff | git apply -)
+      curl https://github.com/MiguelCompany/memory/commit/6041b31341c323d07611051bb6e1970aabee5532.diff | git apply -)
```
Comment (MiguelCompany) [inline CMakeLists.txt]: I see you are applying my PR foonathan/memory#119. I wonder if we should wait till that one is merged.

**Review by MiguelCompany — DISMISSED** [review]
LGTM, but I would like @nuclearsandwich to trigger a full ROS 2 CI on this

**Comment by nuclearsandwich** [comment]
> LGTM, but I would like @nuclearsandwich to trigger a full ROS 2 CI on this

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/14939/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/9681/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/12615/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/15104/)

**Review by MiguelCompany — DISMISSED** [review]
LGTM, but DCO should be fixed

File: CMakeLists.txt (hunk 2 above)
Comment (richiware) [reply CMakeLists.txt]: Already in foonathan/memory `master` branch

**Comment by richiware** [comment]
👆🏾 DCO fixed

**Review by MiguelCompany — DISMISSED** [review]
LGTM but, again, I would like @nuclearsandwich to run a full CI on this

**Comment by MiguelCompany** [comment]
* Windows [Windows ci](http://jenkins.eprosima.com:8080/job/nightly_foonathan_memory_master_windows/323)
* Linux [Linux ci](http://jenkins.eprosima.com:8080/job/nightly_foonathan_memory_master_linux/314)
* Linux aarch64 [Linux aarch64 ci](http://jenkins.eprosima.com:8080/job/nightly_foonathan_memory_master_linux_aarch64/314)

**Comment by richiware** [comment]
@MiguelCompany Already updated to last version tag.

**Comment by MiguelCompany** [comment]
@nuclearsandwich Would you mind running a Full ROS 2 CI for this?

**Comment by EduPonz** [comment]
Hi @clalancette ,

It seems that @nuclearsandwich is out of office. Could you please point us to the appropriate person to run a full ROS 2 CI on this?

**Comment by nuclearsandwich** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/15535/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/10221/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/13205/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/15749/)

**Review by MiguelCompany — APPROVED** [review]
LGTM!

**Pull request merged**